Added Invincbility Code and Stat Modifier Codes

ReclaimerShawn opened the pull request:
- Added an Invincibility code that doesn't use ASM. 
- Considering a 1-shot kill code already exists and I made an Invincibility code for your team only, I removed my old HP Code.
- Added Stat Modifier codes for HP, Attack, Defense, and Evasion. 
- Multiple versions of the HP code was made to simulate having the HP that in-game bosses had to pair with the "Use Vivosaur" codes.
- Changed FP Code to allow for more than 360FP if you've progressed far enough in the game.

ReclaimerShawn committed:
Added Invincbility Code and Stat Modifier Codes

- Added an Invincibility code that doesn't use ASM. 
- Considering a 1-shot kill code already exists and I made a better Invincibility for the team code, I removed my old HP Code.
- Added Stat Modifier codes for HP, Attack, Defense, and Evasion. 
- Multiple versions of the HP code was made to simulate having the HP that in-game bosses had to pair with the "Use Vivosaur" codes.

iSharingan commented:
the invincibility code can actually be adapted from the non-ASM "Enemy always 1 HP" code right above the code you added. Adding the conditionals `4FFFE240 14000000` and `3FFFE240 15000000` just before the B (pointer) code will ensure the pointer is aimed at the correct area of memory used for battles (and should be applied to all pointer codes to prevent crashes. the ending D2 line closes all conditionals, so you don't need to add D0 lines at the end) and a DC code will shift to the correct starting address for the current HP. After that, a 7 code (in place of the 8 code on the 6th line) can be used to check for 'greater than 0' (meaning there's actually a vivosaur there and not an empty slot), then the HP write can be safely done with the code left on without triggering an ARM exception. The C code should still be the set to the right offset to loop the effect for the remaining slots on the player's team.

ReclaimerShawn committed:
Made Invincibility Code Work Automatically

- Implemented fix for Invincibility code based on iSharingan's advice. Now L+R doesn't need to be pressed.
- Removed ARM Exception bug warning from all of my codes, as after the necessary fixes were made and more testing was done, ARM Exception bugs were shown not to occur,

ReclaimerShawn commented:
I made the Invincibility code work based on your suggestions, so thank you for that! You said to use Codetype 7 in the code. But, I used Codetype 8 instead of Codetype 7 for the conditional, as Codetype 7 is the less than conditional, not the greater than conditional like Codetype 8 is.

If I add any other codes from here, it'll be to use the unused Paleo pals. Other than that, I can't think of anymore codes I could possibly create.

iSharingan commented:
Right right. I was going to say A code for not equal, but changed it last second without double checking code types. Still the Vivosaur Stat Codes use pointers and would benefit from the conditionals `4FFFE240 14000000` and `3FFFE240 15000000`. After that we're good to go (I'd add them myself, but it seems to be committing you your repo and not this PR)

ReclaimerShawn committed:
Add Conditionals To All Stat Modifier

ReclaimerShawn commented:
Added the conditionals into all of the pointer codes.
